robottelo/config/base.py

```diff
@@ -78,10 +81,17 @@ def get(self, section, option, default=None, cast=None):
             defined.
         :param cast: If provided the value will be cast using the cast
             provided.
-
         """
+        # First try to read from environment variable.
+        # [bugzilla]
+        # api_key=123456
+        # can be expressed as:
+        # $ export ROBOTTELO_BUGZILLA_API_KEY=123456
+        value = os.environ.get(f'ROBOTTELO_{section.upper()}_{option.upper()}')
```

Comment:
we could just pull everything prefixed with ROBOTTELO_. If something we're not expecting is already there, then it can still be added to the runtime config and never used.

Reply:
@JacobCallahan then I recommend the use of http://dynaconf.readthedocs.io

Reply:
@JacobCallahan since both section and option can have any number of underscores, I don't think there is generic way to correctly extract these out of arbitrary string. At least, you need to compare section name against list of known names. I'm afraid this will be much more code than literally two lines added here.

And then, I don't see a difference between "never added from environment" (this PR) and "added from environment, but never retrieved (since nothing references it)" (your proposal, as I understand it).

If there is no visible difference between this and your propsal, and this PR achieves it in smaller and simpler code, what would be benefit of your proposal?

Reply:
@mirekdlugosz I do something similar with plinko [1] where I just pull everything that has a pre-defined prefix. In plinko's case, it is PLINKO_. If we wanted it to be compatible with other presets, we could simply strip the "ROBOTTELO_", make the rest lowercase, and apply it to the running config. No need to get overly complicated.

Reply:
@rochacbruno i'm just thinking about the possibility of a setting not conforming to the ROBOTTELO_X_Y naming convention. Not a blocker at all, just some ideas to make it more flexible.

Reply:
@JacobCallahan I am ok to make it more flexible, but I am concerned this will require too much complexity right now with no immediate advantage (use case). to do that we need to avoid conflicts with existing sections on properties file.

I suggest we do that in a different PR, and by adding a `meta` section to the config like `settings.env` and then we can access arbitrary variables only via that endpoint.

```bash
export ROBOTTELO_FOO_BAR=1
export ROBOTTELO_BUGZILLA_API_KEY=nskfnsakjnfajskf
```

then

```py
settings.env.foo_bar == 1
settings.env.bugzilla_api_key == 'nskfnsakjnfajskf'
settings.bugzilla.api_key == 'nskfnsakjnfajskf'
```
`settings.env` will be a catchall for all existing `ROBOTTELO_` envvars.